glib/src/gstring.rs

```diff
@@ -404,6 +404,48 @@ impl<'a> ToGlibPtr<'a, *mut c_char> for GString {
     }
 }
 
+#[doc(hidden)]
+impl<'a> FromGlibContainer<*const c_char, *const i8> for GString {
+    unsafe fn from_glib_none_num(ptr: *const i8, num: usize) -> Self {
+        if num == 0 {
+            return Self::from("");
+        }
+        let mut bytes = Vec::with_capacity(num);
+
+        for i in 0..num {
+            let b = std::ptr::read(ptr.add(i)) as u8;
+            bytes.push(b);
+        }
+        CStr::from_bytes_with_nul_unchecked(bytes.as_slice()).into()
+    }
+
+    unsafe fn from_glib_container_num(ptr: *const i8, num: usize) -> Self {
```

Comment:
`container` and `full` could simply create a `GString(Inner::Foreign(ptr as *mut _, num))` without copying or anything

Reply:
I've added a very simple test just to check if everything was working fine but the `container` & `full` one fails with a `free(): invalid pointer`, I wonder whether what I wrote is just completely wrong on that test or something isn't working properly? :)

Reply:
See comments below